### PR TITLE
Tracing context for log4j v1.2

### PR DESCRIPTION
**Pull request by garyd203**

### CODE REVIEW

File: context/log4j12/pom.xml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <parent>
+    <groupId>io.zipkin.brave</groupId>
+    <artifactId>brave-context-parent</artifactId>
+    <version>4.3.0-SNAPSHOT</version>
```
Comment (adriancole) [inline context/log4j12/pom.xml]: 4.3.1 :)

File: context/log4j12/src/main/java/brave/context/log4j12/MDCCurrentTraceContext.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    MDCCurrentTraceContext(CurrentTraceContext delegate) {
+        if (delegate == null) {
+            throw new NullPointerException("delegate == null");
+        }
+        this.delegate = delegate;
+    }
+
+    @Override
+    public TraceContext get() {
+        return delegate.get();
```
Comment (adriancole) [inline context/log4j12/src/main/java/brave/context/log4j12/MDCCurrentTraceContext.java]: please use square formatter from CONTRIBUTING, notably 2-space

File: context/log4j12/src/test/java/brave/context/log4j12/MDCCurrentTraceContextTest.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+                .isNull();
+        testLogger.info("no span");
+
+        Tracer tracer = Tracing.newBuilder()
+                .currentTraceContext(MDCCurrentTraceContext.create())
+                .build().tracer();
+
+        Span parent = tracer.newTrace();
+        try (Tracer.SpanInScope wsParent = tracer.withSpanInScope(parent)) {
+            testLogger.info("with span: " + parent);
```
Comment (adriancole) [inline context/log4j12/src/test/java/brave/context/log4j12/MDCCurrentTraceContextTest.java]: maybe add a log4j.xml file in test/resources so you can see this works?

**Comment by adriancole** [comment]
looks good! only nits

File: context/log4j12/pom.xml (hunk 1 above)
Comment (garyd203) [inline context/log4j12/pom.xml]: It was correct when we started two hours earlier :-)

Comment (adriancole) [inline context/log4j12/pom.xml]: hehe

**Comment by adriancole** [comment]
ps lemme know if you want I can polish on the way in (I know it is a friday, so offering :P)

**Comment by garyd203** [comment]
All tidied up, thanks @adriancole

**Pull request merged**